crates/ruff_linter/src/rules/ruff/rules/dict_update_single.rs

```diff
@@ -0,0 +1,127 @@
+use ruff_diagnostics::{AlwaysFixableViolation, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::{DictItem, Expr, ExprAttribute, ExprCall, ExprDict};
+use ruff_python_semantic::analyze::typing;
+use ruff_python_semantic::SemanticModel;
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+
+/// Real type: [`Expr::Name`]
+type DictRef = Expr;
+type Argument = Expr;
+type Key = Expr;
+type Value = Expr;
+
+/// ## What it does
+/// Checks for `dictionary.update({single: item})`.
+///
+/// ## Why is this bad?
+/// Calling `dict.update()` with a single-item dictionary
+/// is equivalent to setting the item directly,
+/// but more verbose and potentially more expensive.
```

Comment:
I suggest removing the mention of performance as discussed in the issue. There's some evidance that directly assigning is faster, but it isn't the rule's primary motivation and it only matters if this code is in a hot loop. 

```suggestion
/// Calling `dict.update()` with a single-item dictionary
/// is equivalent to setting the item directly, which is simpler and more concise. 
```